npm/react/src/createMount.ts

```diff
@@ -136,40 +135,33 @@ export const makeMountFn = (
  * This is designed to be consumed by `npm/react{16,17,18}`, and other React adapters,
  * or people writing adapters for third-party, custom adapters.
  */
-export const makeUnmountFn = (options: UnmountArgs, internalUnmountOptions: InternalUnmountOptions) => {
+export const makeUnmountFn = (options: UnmountArgs) => {
   return cy.then(() => {
-    return cy.get(ROOT_SELECTOR, { log: false }).then(($el) => {
-      if (lastMountedReactDom) {
-        internalUnmountOptions.unmount($el[0])
-        const wasUnmounted = internalUnmountOptions.unmount($el[0])
-
-        if (wasUnmounted && options.log) {
-          Cypress.log({
-            name: 'unmount',
-            type: 'parent',
-            message: [options.boundComponentMessage ?? 'Unmounted component'],
-            consoleProps: () => {
-              return {
-                description: 'Unmounts React component',
-                parent: $el[0],
-                home: 'https://github.com/cypress-io/cypress',
-              }
-            },
-          })
-        }
-      }
-    })
+    const wasUnmounted = mountCleanup()
```

Comment:
Can this be undefined? I'm surprised we don't need a null check, since the definition is `let mountCleanup: InternalMountOptions['cleanup']`

Should we add a null check? Could just be `mountCleanup?.()`.